clients/drcachesim/tracer/raw2trace.h

```diff
@@ -70,6 +71,108 @@ struct module_t {
     bool is_external; // If true, the data is embedded in drmodtrack custom fields.
 };
 
+/**
+ * module_mapper_t maps and unloads application modules.
+ */
+class module_mapper_t final {
+public:
+    /**
+     * Parses and iterates over the list of modules.  This is provided to give the user a
+     * method for iterating modules in the presence of the custom field used by drmemtrace
+     * that prevents direct use of drmodtrack_offline_read().
+     * On success, calls the \p process_cb function for every module in the list, and
+     * returns an empty string at the end. Returns a non-empty error message on failure.
+     */
+    module_mapper_t(const char *module_map_in,
+                    const char *(*parse_cb)(const char *src, OUT void **data) = nullptr,
+                    std::string (*process_cb)(drmodtrack_info_t *info, void *data,
+                                              void *user_data) = nullptr,
+                    void *process_cb_user_data = nullptr,
+                    void (*free_cb)(void *data) = nullptr, uint verbosity_in = 0);
+
+    /**
+     * All APIs on this type, including constructor, may fail. get_last_error() returns
+     * the last error message. The object should be considered unusable if
+     * !get_last_error().empty()
+     */
+    std::string
+    get_last_error(void) const
+    {
+        return last_error;
+    }
+
+    /**
+     * module_t vector corresponding to the application modules. Lazily loads and caches
+     * modules. If the object is invalid, returns an empty vector.
+     */
+    const std::vector<module_t> &
+    get_loaded_modules()
+    {
+        if (last_error.empty() && modvec.empty())
+            read_and_map_modules();
+        return modvec;
+    }
+
+    /**
+     * This interface is meant to be used with a final trace rather than a raw
+     * trace, using the module log file saved from the raw2trace conversion.
+     * After the a call to get_loaded_modules(), this routine may be used
```

Comment:
This is confusing to the user: the user is likely going to call raw2trace_t::do_module_parsing_and_mapping() and never directly call get_loaded_modules().  Please clarify: may require elaboration here and in raw2trace_t::do_module_parsing_and_mapping() and raw2trace_t::find_mapped_trace_address().

Reply:
done, but might be moot (see 2 comment threads down)